### PR TITLE
Refactor test suite with factory fixtures and feature constants

### PR DESCRIPTION
**Pull request by cmalinmayor**
This comprehensive refactor improves test maintainability and reduces code duplication across the test suite while maintaining 100% test coverage (181 tests passing).

### Added Feature Constants (conftest.py)
- FEATURES_WITH_SEG = ["pos", "area", "iou"]
- FEATURES_NO_SEG = ["pos"]
- SOLUTION_FEATURES_WITH_SEG = ["pos", "area", "iou", "tracklet_id"]
- SOLUTION_FEATURES_NO_SEG = ["pos", "tracklet_id"]
- Ensures consistency across all test files and eliminates magic lists

### Added Factory Fixtures
- get_tracks(ndim, with_seg, is_solution): Unified tracks creation
  - Automatically selects correct graph and segmentation fixtures
  - Automatically applies appropriate feature constants
  - Deep copies graphs to prevent fixture pollution
- get_graph(ndim, with_features): Simplified graph selection
  - with_features options: "clean", "position", "computed"
  - Eliminates 15+ instances of manual graph fixture selection
- get_segmentation(ndim): Simplified segmentation selection
  - Eliminates 12+ instances of conditional segmentation logic

### Refactored with Factory Fixtures
- All action tests (test_add_delete_edge, test_add_delete_nodes, etc.)
- All annotator tests (43 tests across 4 test files)
- All user action tests
- Import/export tests (test_export_to_geff, test_internal_format)
- Data model tests (test_solution_tracks, test_tracks)

### Code Patterns Eliminated
- "seg = segmentation_2d if ndim == 3 else segmentation_3d" (12+ instances)
- Manual graph fixture selection based on ndim (15+ instances)
- Inconsistent existing_features specifications

## Source Code Changes
- Fixed edge case in add_delete_edge.py for proper edge removal handling
- Added validation in add_delete_node.py for node attribute consistency

## Impact
- Improved test readability and maintainability
- Reduced code duplication by ~100 lines
- Centralized test configuration in conftest.py
- Easier to add new tests following established patterns
- All 181 tests passing, 6 skipped (intentional)

### CODE REVIEW

File: src/funtracks/actions/add_delete_node.py [hunk 1]
```diff
@@ -79,6 +79,12 @@ def _apply(self) -> None:
         for attr, values in attrs.items():
             self.tracks._set_node_attr(self.node, attr, values)
 
+        # track_id is required for SolutionTracks
```
Comment (cmalinmayor) [inline src/funtracks/actions/add_delete_node.py]: TODO for later: put all validation in the inits, and update the docstrings with the exceptions, for all the actions
